doc/cephadm/services/index.rst

```diff
@@ -541,13 +541,60 @@ a spec like
 
 which would cause each mon daemon to be deployed with `--cpus=2`.
 
+There are two ways to express arguments in the ``extra_container_args`` list.
+To start, an item in the list can be a string. When passing an argument
+as a string and the string contains spaces, Cephadm will automatically split it
+into multiple arguments. For example, ``--cpus 2`` would become ``["--cpus",
+"2"]`` when processed. Example:
```

Comment:
While this bit about how it becomes `["--cpus", "2"]` is all accurate, for user facing docs, do we think this will be clear as to what this ends up becoming for the actual daemon/container we deploy or should we be explaining that?

Reply:
That's a tough one. Honestly, it is sometimes difficult to explain how *argv* works to some programmers. This is fairly precise and short - I can't think of any other way to express it that is precise and not verbose. We can drop those sentences entirely... but that might lose a little clarity.

Reply:
I can't really think of any great options. It's hard to express that `--cpus 2` becomes `["--cpus", "2"]` which then becomes `--cpus 2` again when we generate the cli command to run the container at the end and how that is different from our other non-split case where `--cpus 2` would become `['--cpus 2']` which actually formerly just broke when passed to cephadm, hence the original desire to split it, but now is a valid way to express special args that need spaces in them. Given how hard it is to do that nicely, I won't press for me here. This description is probably good enough and hopefully the examples make it clear that is just a valid format of specifying args (which is the important thing to the user in the end)

Reply:
Not to belabor things, but I disagree with the part "which then becomes --cpus 2 again. But I get that's how a lot of folks might think about it. As of this PR (including needed features from the new deploy) it should never become a single string ever again. The CLI arguments are retained as an "array" until we construct the command line for the unit.run bash file, and the functions for building that file correctly handles using quoting (via the shlex module) to keep from breaking up arguments. The association between argv[N] and argv[N+1] is a purely conventional thing and is dependent on the library doing the CLI parsing.  I'm sorry if this sounds nitpicky but this is topic I care a lot about and can even lead to security issues in some cases, like the early "jailbreaks" of linksys router UIs back in the day.